src/jimp2/wireworld/z8/datamangment/FactoryOfCustomElements.java

```diff
@@ -42,67 +37,74 @@ private void initializeElements() throws IOException {
             files = new String[listOfFiles.length];
 
             for (int i = 0; i < listOfFiles.length; i++) {
-
                 files[i] = listOfFiles[i].getName();
-                FileReader fileReader = null;
-                BufferedReader reader = null;
+                FileReader fileReader;
+                BufferedReader reader;
                 try {
-                    fileReader = new FileReader("CustomElements/"+files[i]);
+                    fileReader = new FileReader(folderName+"/"+files[i]);
                     reader = new BufferedReader(fileReader);
                 } catch (FileNotFoundException e) {
                     System.err.println("There is problem with reading "+files[i]+" file in CustomElement");
                     continue;
                 }
 
 
+                int width;
+                int height;
+                Point inConnectorPoint;
+
+                String line;
+                String [] numbers;
+
                 //reading width and height of custom element
-                String line = reader.readLine();
-                String[] w = line.split("\\s+");
-                if (w.length == 2) {
-                    width = Integer.parseInt(w[0]);
-                    height = Integer.parseInt(w[1]);
+                line = reader.readLine();
+                numbers = line.split("\\s+");
+                if (numbers.length == 2) {
+                    width = Integer.parseInt(numbers[0]);
+                    height = Integer.parseInt(numbers[1]);
                 } else {
                     System.err.println("Wrong data in custom element file (width,height)");
+                    continue;
                 }
 
                 //reading input point of custom element
                 line = reader.readLine();
-                w = line.split("\\s+");
-                if (w.length == 2)
-                    inConnectorPoint = new Point(Integer.parseInt(w[0]), Integer.parseInt(w[1]));
-                else
+                numbers = line.split("\\s+");
+                if (numbers.length == 2)
+                    inConnectorPoint = new Point(Integer.parseInt(numbers[0]), Integer.parseInt(numbers[1]));
+                else {
                     System.err.println("Wrong data in custom element file (input point)");
-
-                CustomElement cs = new CustomElement(width,height,files[i],inConnectorPoint);
+                    continue;
+                }
+                CustomElement customElement = new CustomElement(width,height,files[i],inConnectorPoint);
 
                 //reading table
                 int row = 0;
                 while((line = reader.readLine()) != null) {
-                    w = line.split("\\s+");
-                    if (w.length != width)
+                    numbers = line.split("\\s+");
+                    if (numbers.length != width) {
                         System.err.println("There is not enough points (width)");
+                        continue;
```

Comment:
ten continue to za mało - i tak zapisze ten element w available elements; lepiej będzie break i jakiś bolean że nie zapisujemy albo coś